### PR TITLE
feat: Allow persisit user-defined catalog in metasrv

### PR DESCRIPTION
**Pull request by Xuanwo**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

feat: Allow persisit user-defined catalog in metasrv

Part of https://github.com/datafuselabs/databend/issues/11947

### CODE REVIEW

File: src/meta/api/src/schema_api_impl.rs [hunk 1]
```diff
@@ -2990,6 +2995,82 @@ impl<KV: kvapi::KVApi<Error = MetaError>> SchemaApi for KV {
...
+        &self,
+        req: CreateCatalogReq,
+    ) -> Result<CreateCatalogReply, KVAppError> {
+        debug!(req = debug(&req), "SchemaApi: {}", func_name!());
+
+        let name_key = &req.name_ident;
+
+        let mut retry = 0;
+        while retry < TXN_MAX_RETRY_TIMES {
+            retry += 1;
```
Comment (drmingdrmer) [inline src/meta/api/src/schema_api_impl.rs:3008]: Could be better with:

```rust
        let ctx = &func_name!();
        let mut trials = txn_trials(None, ctx);
        let id = loop {
            trials.next().unwrap()?;
```

**Review by drmingdrmer — APPROVED** [review]

File: src/meta/api/src/schema_api_impl.rs (hunk 1 above)
Comment (Xuanwo) [reply src/meta/api/src/schema_api_impl.rs]: Added.

**Comment by Xuanwo** [comment]
askbend:summary

**Comment by drmingdrmer** [comment]
### PR summary(by Claude):

 * The main changes are:

1. **Add catalog support**  
   Added catalog support, including catalog proto definitions, catalog APIs in meta service, and catalog usage in query service.

2. **Enhance metadata compatibility**
   Enhanced metadata compatibility: proto definitions, from_to_protobuf conversions, and backward compatibility tests.

3. **Misc changes** 
   Minor refactors and bug fixes.

Overall, this pull request adds the basic catalog support which lays the foundation 
for further development on catalogs.

**Pull request merged**